src/termdict/sstable_termdict/sstable/sstable_index.rs

```diff
@@ -41,10 +44,40 @@ pub struct SSTableIndexBuilder {
     index: SSTableIndex,
 }
 
+/// Given that left < right,
+/// mutates `left into a shorter string left'` that
+/// matches `left <= left' < right`.
+fn find_shorter_str_in_between(left: &mut Vec<u8>, right: &[u8]) {
+    assert!(&left[..] < right);
+    let common_len = common_prefix_len(&left, right);
+    if left.len() == common_len {
+        return;
+    }
+    // It is possible to do one character shorter in some case,
+    // but it is not worth the extra complexity
+    for pos in (common_len + 1)..left.len() {
+        if left[pos] != 255u8 {
```

Comment:
what's the meaning of 255 here?

the comment says string, but we operate on bytes, what about utf-8?

Reply:
255 is the max value for a u8.

It is not especially utf8. All of these are byte strings.